apps/wallet-dashboard/app/(protected)/assets/page.tsx

```diff
@@ -31,21 +31,22 @@ export default function AssetsDashboardPage(): React.JSX.Element {
     const account = useCurrentAccount();
     const { data, isFetching, fetchNextPage, hasNextPage, refetch } = useGetOwnedObjects(
         account?.address,
-        undefined,
+        {
+            MatchNone: [{ StructType: '0x2::coin::Coin' }],
+        },
         OBJECTS_PER_REQ,
     );
 
-    const assets: IotaObjectData[] = [];
+    const visualAssets: IotaObjectData[] = [];
+    const otherAssets: IotaObjectData[] = [];
```

Comment:
we simplified this code by removing `visualAssets` and `otherAssets` based on the comment in this PR > https://github.com/iotaledger/iota/pull/4038#discussion_r1856058919
with this PR we would be reverting back the changes

Reply:
That code was incorrect because the objects were always pushed to assets array
`if (selectedCategory == AssetCategory.Visual) {
                    if(hasDisplayData(asset)) {
                        assets.push(asset.data);
                    }
                } else if(selectedCategory == AssetCategory.Other) {
                    assets.push(asset.data);
                }`
I have added a Map to simplify the code, but also to be able to separate it into the 2 categories, take a look and tell me what you think

Reply:
i think that it was correct because only one array is shown at a time. that means that everything can be stored in the same array

Reply:
I don't think so, since before the NFTs were shown with the others filter
![Captura desde 2024-12-20 09-29-29](https://github.com/user-attachments/assets/e1bf11a2-2dd6-4c6e-9bab-d2b95f69177f)